src/main/java/application/controllers/ProductController.java

```diff
@@ -27,6 +27,20 @@ public String createProductForm(Model model) {
         return "createproduct";
     }
 
+    @GetMapping("/filterproduct")
+    public String filterProduct(Model model, @RequestParam(value = "categorySelect") String category) {
+        System.out.println(category.toString());
```

Comment:
Can anyone help me find out why this line prints "CLOTHING" no matter what category you select in the dropdown?

Reply:
Have you tried changing the parameter type from String to Category?

Reply:
ya that throws an error, it is passed as a string

Reply:
I found the issue. I was comparing the label with the enum name itself instead of the enum label.